### ISSUE TITLE
[flaky test][pkg/stanza/fileconsumer] TestNoLostPartial failure

### ISSUE THREAD

**Issue by crobert-1** [open-telemetry/opentelemetry-collector-contrib#35007]
### Component(s)

pkg/stanza/fileconsumer

### Describe the issue you're reporting

[Failing CI/CD link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10699296117/job/29661153229?pr=34513)

Failure output:
```
=== Failed
=== FAIL: fileconsumer TestNoLostPartial (1.10s)
    file_test.go:1474: unexpected token from file "1530210957.log2": aaaaaaaaaaa
    file_test.go:1463: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:1463
        	Error:      	Condition never satisfied
        	Test:       	TestNoLostPartial

=== FAIL: fileconsumer TestEncodings/Nop (unknown)
panic: Fail in goroutine after TestNoLostPartial has completed

goroutine 6697 [running]:
testing.(*common).Fail(0xc0003e1380)
	/opt/hostedtoolcache/go/1.23.0/x64/src/testing/testing.go:953 +0x194
testing.(*common).Errorf(0xc0003e1380, {0x13cca8e, 0x3}, {0xc00011f4d0, 0x1, 0x1})
	/opt/hostedtoolcache/go/1.23.0/x64/src/testing/testing.go:1076 +0xaf
github.com/stretchr/testify/assert.Fail({0x153a6c0, 0xc0003e1380}, {0x13e995a, 0x1d}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:363 +0x447
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest.(*Sink).NextCall(0xc000502018, 0xc0003e1380)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest/sink.go:97 +0x13f
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestNoLostPartial.func1()
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:1467 +0x16d
github.com/stretchr/testify/assert.Eventually.func1()
ERROR rerun aborted because previous run had a suspected panic and some test may not have run
make[2]: *** [../../Makefile.Common:128: test] Error 3
make[1]: *** [Makefile:187: pkg/stanza] Error 2
make: *** [Makefile:127: gotest] Error 2
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x34
created by github.com/stretchr/testify/assert.Eventually in goroutine 165
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x3d6

=== FAIL: fileconsumer TestDefaultBehaviors (unknown)

=== FAIL: fileconsumer TestEncodings/SmileyFaceUTF16 (unknown)

=== FAIL: fileconsumer TestEncodings/SmileyFaceNewlineUTF16LE (unknown)

=== FAIL: fileconsumer TestEncodings/ChineseCharacterBig5 (unknown)

=== FAIL: fileconsumer TestBuildWithSplitFunc/BadExcludeGlob (unknown)

=== FAIL: fileconsumer TestNopEncodingDifferentLogSizes/slightly_smaller_log_size (unknown)

=== FAIL: fileconsumer TestEncodings/InvalidUTFReplacement (unknown)

=== FAIL: fileconsumer TestBuildWithSplitFunc (unknown)

=== FAIL: fileconsumer TestNopEncodingDifferentLogSizes (unknown)
--- PASS: TestNopEncodingDifferentLogSizes/same_size (0.20s)
--- PASS: TestNopEncodingDifferentLogSizes/massive_log_size (0.20s)
--- PASS: TestNopEncodingDifferentLogSizes/slightly_larger_log_size (0.20s)

=== FAIL: fileconsumer TestSymlinkedFiles (unknown)

=== FAIL: fileconsumer TestBuildWithSplitFunc/Basic (unknown)

=== FAIL: fileconsumer TestBuildWithSplitFunc/BadIncludeGlob (unknown)

=== FAIL: fileconsumer TestEncodings (unknown)
panic: Fail in goroutine after TestNoLostPartial has completed

goroutine 6697 [running]:
testing.(*common).Fail(0xc0003e1380)
	/opt/hostedtoolcache/go/1.23.0/x64/src/testing/testing.go:953 +0x194
testing.(*common).Errorf(0xc0003e1380, {0x13cca8e, 0x3}, {0xc00011f4d0, 0x1, 0x1})
	/opt/hostedtoolcache/go/1.23.0/x64/src/testing/testing.go:1076 +0xaf
github.com/stretchr/testify/assert.Fail({0x153a6c0, 0xc0003e1380}, {0x13e995a, 0x1d}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:363 +0x447
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest.(*Sink).NextCall(0xc000502018, 0xc0003e1380)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/emittest/sink.go:97 +0x13f
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer.TestNoLostPartial.func1()
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/file_test.go:1467 +0x16d
github.com/stretchr/testify/assert.Eventually.func1()
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x34
created by github.com/stretchr/testify/assert.Eventually in goroutine 165
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.9.0/assert/assertions.go:1902 +0x3d6

=== FAIL: fileconsumer TestEncodings/SmileyFaceNewlineUTF16 (unknown)

=== FAIL: fileconsumer TestEncodings/ValidUTF8 (unknown)

=== FAIL: fileconsumer TestBuildWithSplitFunc/InvalidEncoding (unknown)

=== FAIL: fileconsumer TestEncodings/ChineseCharacter (unknown)

DONE 3[301](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10699296117/job/29661153229?pr=34513#step:8:302) tests, 19 failures in 60.522s
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
```

**Comment by crobert-1** [comment]
Slightly different stack, but same test is failing: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/10703836169/job/29675437183?pr=34998

This one shows a data race though.

**Comment by djaglowski** [comment]
Looking into this

**Comment by djaglowski** [comment]
This is easily reproducibly on my local machine. According to git bisect it began with #34888. I don't see an obvious cause though. Continuing to look into it..

**Comment by djaglowski** [comment]
Closing this as resolved by #35012. 

#35014 tracks potential additional work.
